From ae8d4093a25bcd4aa4abb2164702bded5350bf5b Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Wed, 2 Aug 2017 15:04:02 +0800
Subject: [PATCH 2413/5242] MLK-16065-16 usb: cdns3: gadget: fix no one
 handled interrupt issue

commit  3438298d8e16ed0a242073123745a96cd31be66e from
https://source.codeaurora.org/external/imx/linux-imx.git

If there are too many interrupts for non-control ep, the
no-one handled interrupt issue will occur due to without
return IRQ_HANDLED for them.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index 5bf1409..09416c3 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -1183,6 +1183,7 @@ static irqreturn_t cdns_irq_handler_thread(struct cdns3 *cdns)
 		cdns_check_ep_interrupt_proceed(
 				usb_ss->eps[CAST_EP_REG_POS_TO_INDEX(bit_pos)]);
 		reg &= ~bit_mask;
+		ret = IRQ_HANDLED;
 	} while (reg);
 
 irqend:
-- 
1.7.9.5

